import: import worker failure can succeed with missing data

Forked from: https://github.com/cockroachdb/cockroach/issues/102839

```test logs left over in: /tmp/_tmp/060246e25d27a0efc6dd14ec672a0f2b/logTestImportWorkerFailure3155831314
--- FAIL: TestImportWorkerFailure (17.54s)
    test_log_scope.go:161: test logs captured to: /tmp/_tmp/060246e25d27a0efc6dd14ec672a0f2b/logTestImportWorkerFailure3155831314
    test_log_scope.go:79: use -show-logs to present logs inline
    import_stmt_test.go:5435: query 'SELECT * FROM t ORDER BY i': expected:
        0
        1
        2
        3
        4
        5
        6
        7
        8
        9
        10
        11
        12
        13
        14
        15
        16
        17
        18
        19
        
        got:
        1
        2
        4
        5
        7
        8
        10
        11
        13
        14
        16
        17
        19
        
    panic.go:522: -- test log scope end --
FAIL
I230509 15:29:22.694464 1 (gostd) testmain.go:254  [-] 1  Test //pkg/sql/importer:importer_test exited with error code 1
```

Jira issue: CRDB-30515

--- a/pkg/sql/importer/read_import_base.go
+++ b/pkg/sql/importer/read_import_base.go
@@ -112,29 +112,26 @@ func runImport(
 	var summary *kvpb.BulkOpSummary
 	group.GoCtx(func(ctx context.Context) error {
 		summary, err = ingestKvs(ctx, flowCtx, spec, progCh, kvCh)
-		if err != nil {
-			return err
-		}
-		var prog execinfrapb.RemoteProducerMetadata_BulkProcessorProgress
-		prog.ResumePos = make(map[int32]int64)
-		prog.CompletedFraction = make(map[int32]float32)
-		for i := range spec.Uri {
-			prog.CompletedFraction[i] = 1.0
-			prog.ResumePos[i] = math.MaxInt64
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case progCh <- prog:
-			return nil
-		}
+		return err
 	})
 
 	if err = group.Wait(); err != nil {
 		return nil, err
 	}
 
-	return summary, nil
+	var prog execinfrapb.RemoteProducerMetadata_BulkProcessorProgress
+	prog.ResumePos = make(map[int32]int64)
+	prog.CompletedFraction = make(map[int32]float32)
+	for i := range spec.Uri {
+		prog.CompletedFraction[i] = 1.0
+		prog.ResumePos[i] = math.MaxInt64
+	}
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case progCh <- prog:
+		return summary, nil
+	}
 }
 
 type readFileFunc func(context.Context, *fileReader, int32, int64, chan string) error

